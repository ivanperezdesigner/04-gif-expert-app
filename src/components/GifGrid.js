import React, { useEffect, useState } from 'react'
import { GifGridItem } from './GifGridItem.js'
import { getGift } from '../helpers/getGifs.js'

export const GifGrid = ({ category }) => {

    const [images, setImages] = useState([])

    useEffect(() => {
        getGift(category)
            .then(imgs => setImages( imgs))
    }, [category])

    return (
        <>
            <h3>{ category }</h3>
            <div className='card-grid'>
                {
                    images.map(img => (
                    <GifGridItem 
                        key={img.id}
                        {...img}
                    />
                    ))
                }
            </div>
        </>
    )
}
