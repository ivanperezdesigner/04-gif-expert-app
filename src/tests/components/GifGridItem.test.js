import { shallow } from 'enzyme'
import { GifGridItem } from '../../components/GifGridItem'
import React from 'react'

describe('Pruebas en <GifGridItem />', () => {

    const title = 'un título'
    const url = 'https://localhost/image.png'
    const wrapper = shallow(<GifGridItem title={title} url={url} /> )

    test('Debe mostrar el componente correctamente', () => {
        expect(wrapper).toMatchSnapshot()
    })

    test('Debe mostrar un párrafo con el titulo', () => {
        const p = wrapper.find('p')
        expect( p.text().trim() ).toBe(title)
    })

    test('Debe tener la imagen igual al url y alt de los props', () => {
        const img = wrapper.find('img')
        // console.log(img.props())
        expect(img.prop('src')).toBe(url)
        expect(img.prop('alt')).toBe(title)
    })
    
    test('Debe tener la clase animate__flipInX', () => {
        const div = wrapper.find('div')
        // console.log(div.prop('className'))
        const className = div.prop('className')
        expect(className.includes('animate__flipInX')).toBe(true)
    })
})
