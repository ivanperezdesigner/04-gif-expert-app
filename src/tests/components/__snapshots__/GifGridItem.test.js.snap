// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> Debe mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__flipInX"
>
  <img
    alt="un título"
    src="https://localhost/image.png"
  />
  <p>
    un título
  </p>
</div>
`;
